libgoal/transactions.go

```diff
@@ -685,6 +685,27 @@ func (c *Client) MakeUnsignedAssetConfigTx(creator string, index uint64, newMana
 	return tx, nil
 }
 
+// MakeUnsignedAssetOptinTx creates a tx template for opting in to assets.
+// Does so by sending a zero amount to self.
+//
+// Call FillUnsignedTxTemplate afterwards to fill out common fields in
+// the resulting transaction template.
+func (c *Client) MakeUnsignedAssetOptinTx(index uint64, account string) (transactions.Transaction, error) {
```

Comment:
nit: you could also call  MakeUnsignedAssetSendTx with correct parameters for opt in to reduce repeated code.

Reply:
Yeah I've updated to do this. Originally I was having some issues using this because I wasn't providing the correct empty argument for clawback address which was causing failures.